src/com/palmergames/bukkit/towny/command/TownCommand.java

```diff
@@ -3978,14 +3978,19 @@ public static void parseTownUnclaimCommand(Player player, String[] split) {
 			}
 		}
 	}
-
+	
 	public static void parseTownMergeCommand(Player player, String[] args) {
+		parseTownMergeCommand(player, args, false, null);
+	}
+
+	public static void parseTownMergeCommand(Player player, String[] args, boolean admin, @Nullable Town town) {
 		Resident resident = TownyUniverse.getInstance().getResident(player.getUniqueId());
-		if (!resident.isMayor()) {
+		if (!resident.isMayor() && !admin) {
 			TownyMessaging.sendErrorMsg(player, Translatable.of("msg_town_merge_err_mayor_only"));
 			return;
 		}
-		Town remainingTown = TownyAPI.getInstance().getResidentTownOrNull(resident); // The isMayor() test guarantees we get a town.		
+		if (admin && town == null) return;

```

Comment:
This should return some feedback to the sender, unless we're certain that town will never be null when sent via admin.

Reply:
I'm just gonna remove the admin variable, I can just use `town != null` instead.